docs/src/HowToGuides/Numerics/DGMethods/how_to_make_a_balance_law.md

```diff
@@ -35,44 +35,44 @@ the following methods, which are computed locally at each nodal point:
+| [`integral_load_auxiliary_state!`](@ref) | specify how to compute integrands. can be functions of the conservative state and auxiliary variables. |
+| [`integral_set_auxiliary_state!`](@ref) | specify which auxiliary variables are used to store the output of the integrals. |
+| [`reverse_integral_load_auxiliary_state!`](@ref) | specify auxiliary variables need their integrals reversed. |
+| [`reverse_integral_set_auxiliary_state!`](@ref) | specify which auxiliary variables are used to store the output of the reversed integrals. |
+| [`update_auxiliary_state!`](@ref) | perform any updates to the auxiliary variables needed at the beginning of each time-step. Can be used to solve non-linear equations, calculate integrals, and apply filters. |
+| [`update_auxiliary_state_gradient!`](@ref) | same as above, but after computing gradients and gradient fluxes in case these variables are needed during the update. |
 
 
 ## Methods to compute fluxes and sources
 | **Method** | Purpose |
 |:-----|:-----|
-| [`flux_first_order!`](@ref DGMethods.flux_first_order!) |  specify `F_{first_order}` for each conservative state variable. can be functions of the conservative state and auxiliary variables. |
-| [`flux_second_order!`](@ref DGMethods.flux_second_order!)    |  specify `F_{second_order}` for each conservative state variable. can be functions of the conservative state, gradient flux state, and auxiliary variables. |
-| [`source!`](@ref DGMethods.source!)            |  specify `S_{non_conservative}`  for each conservative state variable. can be functions of the conservative state, gradient flux state, and auxiliary variables. |
+| [`flux_first_order!`](@ref) |  specify `F_{first_order}` for each conservative state variable. can be functions of the conservative state and auxiliary variables. |
+| [`flux_second_order!`](@ref)    |  specify `F_{second_order}` for each conservative state variable. can be functions of the conservative state, gradient flux state, and auxiliary variables. |
+| [`source!`](@ref)            |  specify `S_{non_conservative}`  for each conservative state variable. can be functions of the conservative state, gradient flux state, and auxiliary variables. |
 
 ## Methods to compute numerical fluxes
 | **Method** | Purpose |
 |:-----|:-----|
-| [`wavespeed`](@ref DGMethods.wavespeed) | specify how to compute the local wavespeed if using the `RusanovNumericalFlux`. |
-| [`boundary_state!`](@ref DGMethods.boundary_state!) | define exterior nodal values of the conservative state and gradient flux state used to compute the numerical boundary fluxes. |
+| [`wavespeed`](@ref) | specify how to compute the local wavespeed if using the `RusanovNumericalFlux`. |
+| [`boundary_state!`](@ref) | define exterior nodal values of the conservative state and gradient flux state used to compute the numerical boundary fluxes. |
```

Comment:
I think this functions are pretty integrally tied to the DG discretization as they are used for computing numerical fluxes. `

boundary_state!` should probably be replaced with a `boundary_condition!` interface of which `boundary_state!` is the DG specific implementation.

the `wavespeed` function here is not a generic wavespeed but a particular weighting coefficient that is representative of a wavespeed used when calculating the rusanov flux. maybe this could have a name separate from `wavespeed` to distinguish from the physical wavespeed (altho that might still be used for the rusanov flux in some cases)

Reply:
I agree with the suggested name changes, I didn't want to make too many changes for this PR.

I would argue that the `wavespeed` is a point-wise method and it may be needed for components in the discretization layer but one can write a spatially continuous form. Also, this method could be used for different spatial discretizations (FDM, FVM), so I don't think it's really DG specific.